packages/@aws-cdk/aws-appmesh/lib/virtual-service.ts

```diff
@@ -51,18 +51,11 @@ export interface VirtualServiceBaseProps {
   readonly virtualServiceName?: string;
 
   /**
-   * The VirtualRouter which the VirtualService uses as provider
+   * The VirtualNode or VirtualRouter which the VirtualService uses as its provider
    *
-   * @default - At most one of virtualRouter and virtualNode is allowed.
+   * @default - No provider
    */
-  readonly virtualRouter?: IVirtualRouter;
-
-  /**
-   * The VirtualNode attached to the virtual service
-   *
-   * @default - At most one of virtualRouter and virtualNode is allowed.
-   */
-  readonly virtualNode?: IVirtualNode;
+  readonly virtualServiceProvider?: VirtualServiceProvider;
```

Comment:
2 changes here:

1. Let's move this property to `VirtualServiceProps` below.
2. Let's remove the `mesh` property from `VirtualServiceProps` (no reason to provide it redundantly when a `VirtualRouter` or `VirtualNode` were provided).

Reply:
1. Do we not need the provider in the base props? How else will we define what type of resource backs the virtual service? 

2. Since the provider is optional, I don't think we can remove the `mesh` property. We still need to know what mesh the VS is created as a part of. We can enforce the mesh properties on the VS and Provider do match.

Reply:
1. We do not, because the provider will be evident from the resource that the `addService` method will be on. For example, for `VirtualNode`, it will be:

```ts
public addService(props: VirtualServiceBaseProps): IVirtualService: {
  return new VirtualService(this, 'VirtualService', {
    ...props,
    provider: ServiceProvider.virtualNode(this),
  });
}
```

And similarly for `Mesh` and `VirtualRouter`.

2. The `provider` should be required, of course 🙂.

Reply:
> Putting in "Request changes" to clear this from my ToDo list. @dfezzie make sure to re-request my review when this is ready!

I ended up getting sidetracked and haven't addressed all the feedback quite yet

Reply:
So I've provided a comment on https://github.com/aws/aws-cdk/issues/9490 that argues we should not have an `addService` method on the `IVirtualNode` and `IVirtualRouter` classes. Will wait on your response before I do anything here.